app/src/main/java/com/example/wizelineandroid/adapter/home/HomeAdapter.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.wizelineandroid.adapter.home
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.example.wizelineandroid.R
+import com.example.wizelineandroid.core.BaseViewHolder
+import com.example.wizelineandroid.data.model.ModelBook
+import com.example.wizelineandroid.databinding.ItemRowBinding
+
+class HomeAdapter(private val booksList: List<ModelBook>, private val itemClickListener: onUserClickListener): RecyclerView.Adapter<BaseViewHolder<*>>() {
```

Comment:
Try to use the new List Adapter
https://ferdinandmarq.medium.com/why-you-should-be-using-the-new-and-improved-listadapter-in-android-17a2ab7ca644